### ISSUE TITLE
A very general question about Train and Test data set.

### ISSUE THREAD

**Issue by gaussEuler** [amueller/introduction_to_ml_with_python#73]
Respected Dr. Muller,
I have a TRAIN set and a TEST set (capital train/test are for original data). The TEST set does not contain the target (y) variable - that is what I have to predict. 
So, at first I only used the TRAIN set. I split the TRAIN into 'train' and 'test' using train_test_split(). Then, I did fit(X_train, y_train), predict(X_test) and score(X_test, y_test) as usual and  performed various evaluations (from the book) to find which model and the parameters to select. So all these was done on the original TRAIN set. 
Once I found the proper model, then I went back to the original TRAIN and TEST set and directly proceeded with predict(X_TEST) to find the unknown target (y_TEST) data.
So, did I do this part correctly? 
I know this is a very general and quite naive question, so I hope you wont mind.

Thank you,
Sincerely,
Nikhilesh

**Comment by amueller** [comment]
Yes, that's correct. If you use the "test" set repeatedly your estimate of how well you'll do on "TEST" will be overly optimistic, though.
